src/openai/azure/_azuremodels.py

```diff
@@ -1,5 +1,58 @@
-from typing import TypedDict
+from typing import List, Optional
+from typing_extensions import TypedDict
+from openai.types.chat import ChatCompletion, ChatCompletionChunk, ChatCompletionMessage
+from openai.types.chat.chat_completion import Choice as ChatChoice
+from openai.types.chat.chat_completion_chunk import ChoiceDelta, Choice as ChatChoiceDelta
+from openai._models import BaseModel
+
 
 class ChatExtensionConfiguration(TypedDict):
     type: str
     parameters: object
+
+
+class ContentFilterResult(BaseModel):
+    severity: str
+    filtered: bool
+
+
+class ContentFilterResults(BaseModel):
+    hate: ContentFilterResult
+    self_harm: ContentFilterResult
+    violence: ContentFilterResult
+    sexual: ContentFilterResult
+
+class PromptFilterResult(BaseModel):
+    prompt_index: int
+    content_filter_results: ContentFilterResults
+
+class AzureChatExtensionsMessageContext(BaseModel):
+    messages: List[ChatCompletionMessage]
+
+
+class AzureChatCompletionMessage(ChatCompletionMessage):
+    context: Optional[AzureChatExtensionsMessageContext] = None
+
+
+class AzureChatCompletionChoice(ChatChoice):
+    content_filter_results: Optional[ContentFilterResults] = None
+    message: AzureChatCompletionMessage  # TODO typing hates this
+
+
+class AzureChatCompletion(ChatCompletion):
+    choices: List[AzureChatCompletionChoice]  # TODO typing hates this
+    # TODO service is still returning prompt_filter_results OR prompt_annotations
+    # prompt_filter_results: Optional[List[PromptFilterResult]] = None  
+    prompt_annotations: Optional[List[PromptFilterResult]] = None
```

Comment:
I think we can try to use typing.Sequence to limit the untyped:ness we expose ourselves to. If we get negative feedback, it will give us an opportunity to better understand *why* someone would want to mutate the return value.

Reply:
Using typing.Sequence in place of typing.List seems to break the deserialization

Reply:
Hmmm... interesting. Pydantic supports sequence for deserialization. So there has to be something else that is going on there?

Reply:
I definitely blamed myself first, but toggling only that is what breaks it. Will need to investigate more...

Reply:
Not urgent.

Reply:
Just to be clear, when I say "breaks" I don't mean that it explodes. I just mean that it returns a dict instead of the model